vcd/datasource_vcd_vapp_vm_test.go

```diff
@@ -0,0 +1,87 @@
+// +build vapp ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+// TestAccVcdVappDS tests a vApp data source if a vApp is found in the VDC
+func TestAccVcdVappVmDS(t *testing.T) {
+	// This test requires access to the vCD before filling templates
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	vapp, err := getAvailableVapp()
+	if err != nil {
+		t.Skip("No suitable vApp found for this test")
+		return
+	}
+	var vm *govcd.VM
+
+	if vapp.VApp.Children != nil && len(vapp.VApp.Children.VM) > 0 {
+		vm, err = vapp.GetVMById(vapp.VApp.Children.VM[0].ID, false)
+		if err != nil {
+			t.Skip(fmt.Sprintf("error retrieving VM %s", vapp.VApp.Children.VM[0].Name))
+			return
+		}
+	}
+
+	var params = StringMap{
+		"Org":      testConfig.VCD.Org,
+		"VDC":      testConfig.VCD.Vdc,
+		"VappName": vapp.VApp.Name,
+		"VmName":   vm.VM.Name,
+		"FuncName": "TestVappVmDS",
+		"Tags":     "vm",
+	}
+	configText := templateFill(datasourceTestVappVm, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckOutput("name", vm.VM.Name),
+					resource.TestCheckOutput("storage_profile", vm.VM.StorageProfile.Name),
+					resource.TestCheckOutput("description", vm.VM.Description),
+					resource.TestCheckOutput("href", vm.VM.HREF),
```

Comment:
shouldn't we check all values are there after read - to support future changes?

Reply:
There is another test that does a thorough check of all properties

Reply:
you mean resource test?

Reply:
yes